devito/ops/transformer.py

```diff
@@ -45,17 +61,61 @@ def opsit(trees, count):
         parameters
     )
 
-    stencil_arrays_initializations = itertools.chain(*[
-        to_ops_stencil(p, node_factory.ops_args_accesses[p])
+    # Extract all symbols that need to be converted to ops_dat
+    to_dat = set()
+    symbols = set(FindSymbols('symbolics').visit(trees[0].root))
+    symbols -= set(FindSymbols('defines').visit(trees[0].root))
+    to_dat |= symbols
+
+    # To ensure deterministic code generation we order the datasets to
+    # be generated (since a set is an unordered collection)
+    to_dat = filter_sorted(to_dat)
+
+    name_to_ops_dat = {}
+    pre_loop = []
+    for f in to_dat:
+        if f.is_Constant:
+            continue
+
+        _ops_dat = create_ops_dat(f, name_to_ops_dat, block)
```

Comment:
Also the ops_dats have already been created in the Operator

Reply:
They aren't. Stencil, dat and ops_par_loop are correlated, so they are all in the same place.

Reply:
They are. See https://github.com/opesci/devito/blob/master/devito/ops/operator.py#L55
`name_to_ops_dat` can then be passed into `opsit` to retrieve them. Check #901

Reply:
Check files changed.

Reply:
> name_to_ops_dat can then be passed into opsit
Seems like a good idea.